modules/pool/lib/pool-gql-loader.service.ts

```diff
@@ -66,6 +68,20 @@ export class PoolGqlLoaderService {
         return pools.map((pool) => this.mapToMinimalGqlPool(pool));
     }
 
+    public async getFeaturedPools(): Promise<GqlFeaturedPool[]> {
+        const poolsMetadata = await poolMetadataService.getPoolsMetadata();
+        const poolIds = poolsMetadata.map((pool) => pool.id);
+
+        const pools = await this.getPools({ where: { idIn: poolIds } });
```

Comment:
This fetches the pools from the subgraph but we want to avoid that. Pools are synced to the database. You can query them like this:

```
    const pools = await prisma.prismaPool.findMany({
        where: { id: { in: poolIds }, chain: networkContext.chain },
    });
```

This returns the prismaPool object, not sure if you need additional data. Also, you need to pass it the chain you request the pools for via network context. Or better, your gql resolver should take a chain array as argument and you query the pools for those chains.

Reply:
https://github.com/beethovenxfi/beethovenx-backend/pull/496/commits/9b5f0f6b5f793991178215c93789c651c7996d7c